server/app/rest_api/routes/connector.py

```diff
@@ -0,0 +1,168 @@
+from typing import List, Optional
+
+from classy_fastapi import Routable, get
+from fastapi import APIRouter, Query, status
+from fastapi.responses import JSONResponse, StreamingResponse
+
+from app.schemas import connector as schemas
+from app.tags import Tags
+
+
+class ConnectorRoutes(Routable):
+    def __init__(self):
+        super().__init__()
+
+    @get(
+        "/metadata/connector",
+        operation_id="get_connector_metadata",
+        name="Get Connector Metadata",
+        tags=[Tags.Data_products],
+        response_model=schemas.ConnectorMetadata,
+    )
+    async def get_connector_metadata(self) -> JSONResponse:
+        return JSONResponse(
+            content={
+                "connector_id": "ds-connector-service",
+                "region": "eu-central-1",
+                "supported_interfaces": ["s3", "rest", "sql"],
+                "status": "healthy",
+                "version": "0.1.0",
+            },
+            status_code=status.HTTP_200_OK,
+        )
+
+    @get(
+        "/metadata/{interface_id}/{resource_path:path}/{resource_name}",
+        operation_id="get_dataproduct_metadata",
+        name="Get Data Product Metadata",
+        tags=[Tags.Data_products],
+        response_model=schemas.DataProduct,
+    )
+    async def get_dataproduct_metadata(
+        self,
+        interface_id: str,
+        resource_path: str,
+        resource_name: str,
+    ) -> JSONResponse:
+        """Return metadata of a single data product (plain JSON)."""
+        product = schemas.DataProduct(
+            id=f"s3://{resource_path}/{resource_name}",
+            identifier=resource_name,
+            title=resource_name,
+            description=f"Description for {resource_name}",
+            publisher={"type": "organization", "name": "ds-connector-service"},
+            keyword=["s3"],
+            distribution=[
```

Comment:
@renuka-gollu I am not sure, I though here we are going to return only the distribution part of the ontology ![alt text](image.png)

Reply:
https://www.w3.org/TR/vocab-dcat-3/